### PR TITLE
Removed get_data and init_data, changed TimeData.initialize()

### PR DESCRIPTION
**Pull request by vincepandolfo**
This PR tries to address some of the problems discussed in issue #71 .

- It is now possible to initialize TimeData objects using an `initializer` function. (Note: This is also the only way to initialize them if `save=True`). `test_save.py` has been modified to demonstrate this.
- Removed `get_data` and `init_data`

### CODE REVIEW

**Comment by FabioLuporini** [comment]
So the only thing that I think we also sort of discussed is that the ``initializer`` function could be provided directly through the constructor. Is this still correct ?

**Comment by mlange05** [comment]
Looks good to me. I would like to add the `initializer` function to the constructor, although that's not a critical requirement.

File: tests/test_save.py [hunk 1]
```diff
@@ -35,9 +39,9 @@ def run_simulation(save=False, dx=0.01, dy=0.01, a=0.5, timesteps=100):
     op.apply()
 
     if save:
-        return u.get_data(timesteps - 2)
+        return u.data[timesteps - 1, :]
     else:
-        return u.get_data()
+        return u.data[0, :]
```
Comment (navjotk) [inline tests/test_save.py]: Do you want to use the `%` operator here to make sure this works even if the number of timesteps above is changed? That would also make the code more self-explanatory.

**Comment by FabioLuporini** [comment]
Now it is in the constructor. Good

**Comment by navjotk** [comment]
Merged

**Pull request merged**